source/data-formats/bson.txt

```diff
@@ -212,6 +216,41 @@ BSON document stored in ``myFile.bson``:
    ``System.IO.Stream`` object. This means that you can read or write any location 
    that can be accessed by a stream.
 
+Implement a Custom BSON Reader and Writer
```

Comment:
I don't think any user would (or should) ever want to implement a custom `IBsonReader` or `IBsonWriter`.

If we want to document anything at all about these interfaces it should be:

1. That `IBsonReader`/`IBsonWriter` interfaces can be used to read "documents" represented in several formats
2. The we have readers and writers for 3 formats: BSON, JSON and in-memory `BsonDocument`

The three sets of standard readers and writers are:

1. `BsonBinaryReader` and `BsonBinaryWriter`
2. `JsonReader` and `JsonWriter`
3. `BsonDocumentReader` and `BsonDocumentWriter`

They all implement the same interfaces.